File: result/translate/starchat/java2python/python/HOW_TO_PRINT_MAXIMUM_NUMBER_OF_A_USING_GIVEN_FOUR_KEYS.py
def f_gold(N):
        if N <= 6:
            return N
        screen = [0] * (N)
        b = 0
        n = 1
        while n <= 6:
            screen[n - 1] = n
            n += 1
        while n <= N:
            screen[n - 1] = max(2 * screen[n - 4], max(3 * screen[n - 5], 4 * screen[n - 6]))
            n += 1
        return screen[N - 1]

File: result/translate/starchat/python2java/java/HOW_TO_PRINT_MAXIMUM_NUMBER_OF_A_USING_GIVEN_FOUR_KEYS.java
public static int f_gold ( int N ) {
        if ( N <= 6 ) return N;
        int[] screen = new int[N];
        for ( int n = 1; n < 7; n++ ) screen[n - 1] = n;
        for ( int n = 7; n < N + 1; n++ ) {
            screen[n - 1] = Math.max( 2 * screen[n - 4], Math.max( 3 * screen[n - 5], 4 * screen[n - 6] ) );
        }
        return screen[N - 1];
    }